books/kestrel/c/syntax/validator.lisp

```diff
@@ -1151,22 +1152,24 @@
         (retok (type-unknown)))
        (type1 (type-apconvert type-arg1))
        (type2 (type-apconvert type-arg2))
-       ((unless (or (and (type-case type1 :pointer)
-                         (type-integerp type2))
-                    (and (type-integerp type1)
-                         (type-case type2 :pointer))))
+       ((when (and (type-case type1 :pointer)
+                   (type-integerp type2)))
+        (retok (type-pointer->to type1)))
+       ((unless (and (type-integerp type1)
+                     (type-case type2 :pointer)))
         (retmsg$ "In the array subscripting expression ~x0, ~
```

Comment:
This should be allowed according to [C17:6.5.2.1]: A[B] is the same as B[A].

Reply:
I believe it is still allowed. The condition of the `unless` is the allowed behavior. (I can change it to two `when`s if the asymmetry is confusing.)

Reply:
Yes, I misread, sorry.